components/screens/CheckOut/AddCreditCard.js

```diff
@@ -0,0 +1,219 @@
+import React, { Component } from 'react'
+import { StyleSheet,View,Text, TouchableOpacity, TextInput,Dimensions,Alert } from 'react-native'
+import { connect } from 'react-redux';
+
+
+
+class AddCreditCard extends Component {
+
+    constructor(props){
+		super(props)
+		this.state = { 
+            creditCardNum : '',
+            expDate : '',
+            holderName : '',
+            cardCVV : '',
+        }
+        this.cardNumberDigits = ''
+	}
+
+    static navigationOptions = {
+        headerTitle: 'Add Card'
+      }
+
+
+    handleCardNumberTextChange(txt) {
+        const nums = ['0','1','2','3','4','5','6','7','8','9']
+
+        let creditCardNum = this.state.creditCardNum
+        let cardNumberDigits = this.cardNumberDigits
+        if (txt.length>creditCardNum.length){
```

Comment:
avoid so much nesting of the logic. Either club the conditions or move some conditions else where.